tests/testthat/test_vibrance.R

```diff
@@ -2,83 +2,96 @@ context("Adjust vibrance/saturation of an image")
-                           14, 15, 5),
-                         c(62, 62, 98,
-                           62, 62, 98, # G channel
-                           62, 62, 98),
-                         c(242, 242, 7,
-                           242, 242, 7, # B channel
-                           242, 242, 7)),
+expected_img1 <- array(c(c(14, 14, 14,
+                           15, 15, 15, # R channel
+                           5, 5, 5),
+                         c(62, 62, 62,
+                           62, 62, 62, # G channel
+                           98, 98, 98),
+                         c(242, 242, 242,
+                           242, 242, 242, # B channel
+                           7, 7, 7)),
+                       dim = c(3, 3, 3))
+
+#expected greyscale image with -10 intensity in vibrance()
+expected_img2 <- array(c(c(128, 128, 128,
+                           128, 128, 128, # R channel
+                           52, 52, 52),
+                         c(128, 128, 128,
+                           128, 128, 128, # G channel
+                           52, 52, 52),
+                         c(128, 128, 128,
+                           128, 128, 128, # B channel
+                           52, 52, 52)),
                        dim = c(3, 3, 3))
 
 
-# test for implementation correctness
+# test for correct vibrance
 
-test_that("Return identical image if intensity is 0", {
+test_that("vibrance of image is correctly enhanced", {
 
-  vibrance("test_img/vibrance/test_img1.png", 0, "test_img/vibrance/vibrance.png")
-  output_img <- readPNG("test_img/vibrance/vibrance.png")
-  expect_equal(output_img, test_img1, tolarance = 1e-5)
+  vibrance("test_img/vibrance/test_img1.png", 5, F, "test_img/vibrance/vibrance.png")
+  output_img <- readPNG("test_img/vibrance/vibrance.png") * 255
+  expect_equal(expected_img1, output_img, tolerance = 1e-5)
 
 })
 
+# test for greyscale vibrance with intensity set to -10
+
 test_that("vibrance of image is correctly enhanced", {
 
-  vibrance("test_img/vibrance/test_img1.png", 5, "test_img/vibrance/vibrance.png")
-  output_img <- readPNG("test_img/vibrance/vibrance.png")
-  expect_equal(output_img, expected_img1, tolarance = 1e-5)
+  vibrance("test_img/vibrance/test_img1.png", -10, F, "test_img/vibrance/vibrance.png")
+  output_img <- readPNG("test_img/vibrance/vibrance.png") * 255
+  expect_equal(expected_img2, output_img, tolerance = 1e-5)
 
 })
 
 # test for exception handling
 
-test_that("Input/output image path should be a string", {
+test_that("Input image path should be a string", {
 
-  expect_error(vibrance(888, 5, "test_img/vibrance/vibrance.png"))
-  expect_error(vibrance("test_img/vibrance/test_img1.png", 5, 888))
+  expect_error(vibrance(888, 5, F, "test_img/vibrance/vibrance.png"))
+  expect_error(vibrance("test_img/vibrance/test_img1.png", 5, F, 888))
 
 })
 
-test_that("Intensity should be an integer between 0 and 10", {
+test_that("Intensity should be between 0 and 10", {
```

Comment:
Here you can say "Intensity should be between -10 and 10" for your particular function. :grinning: 
```suggestion
test_that("Intensity should be between -10 and 10", {
```

Reply:
updated in newest commit